lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
So, you can get what you want now in the current develop branch if you just make sure that the xl compiler entry in your compilers.yaml is pointing to the xlc_r, etc compilers. The only reason we would want to create a new xl_r compiler class is if you believe that a user would realistically want to have both xl and xl_r installations, and that these two should be kept separate. It's really a matter of _how_ different xl and xl_r are.